docs/Serverless.md

```diff
@@ -91,3 +95,96 @@ An example deployable with [claudia.js](https://claudiajs.com/tutorials/serverle
 
 - API Gateway doesn't support streams yet, so you're not able to handle [streams](https://www.fastify.io/docs/latest/Reply/#streams). 
 - API Gateway has a timeout of 29 seconds, so it's important to provide a reply during this time.
+
+## Google Cloud Run
+
+Unlike AWS Lambda or Google Cloud Functions, Google Cloud Run is a serverless **container** environment. It's primary purpose is to provide an infrastucture-abstracted environment to run arbitrary containers. As a result, Fastify can be deployed to Google Cloud Run with little-to-no code changes from the way you would write your Fastify app normally.
+
+*Follow the steps below to deploy to Google Cloud Run if you are already familiar with gcloud or just follow their [quickstart](https://cloud.google.com/run/docs/quickstarts/build-and-deploy)*
+
+### Adjust Fastfiy server
+
+In order for Fastify to properly listen for requests within the container, be sure to set the correct port and address:
+
+```js
+const Fastify = require("fastify")
+
+// Google Cloud Run will set this environment variable for you, so
+// you can also use it to detect if you are running in Cloud Run
+const IS_GOOGLE_CLOUD_RUN = process.env.K_SERVICE !== undefined
+
+// You must listen on the port Cloud Run provides
+const port = process.env.PORT || 3000
+
+// You must listen on all IPV4 addresses in Cloud Run
+const address = IS_GOOGLE_CLOUD_RUN ? "0.0.0.0" : undefined
+
+const fastify = Fastify({ trustProxy: true })
+
+async function start() {
+  const address = await server.listen(port, address)
+
+  console.log(`Listening on ${address}`)
+}
+
+start().catch(err => {
+  console.error(err)
+  process.exit(1)
+})
+```
+
+### Add a Dockerfile
+
+You can add any valid `Dockerfile` that packages and runs a Node app. A basic `Dockerfile`, taken from [here](https://cloud.google.com/run/docs/quickstarts/build-and-deploy) would like the following (assuming you have defined `start` as a script in your `package.json`):
+
+```Dockerfile
```

Comment:
I would remove this part linking directly the docker file in the GH repo.

Someone else agrees?

Reply:
It's more likely that they would need to modify this basic dockerfile anyways (for example, ours uses yarn and includes a .npmrc file for private packages. As a result, I think it's more important to just give a generic template as a base, I wouldn't say that google cloud run needs any specific dockerfile config anyways. It's quite flexible.

Reply:
It makes sense 👍